containers/flatpak/prepare

```diff
@@ -1,83 +1,212 @@
+        'runtime': 'org.gnome.Platform',
+        'runtime-version': '43',
+        'sdk': 'org.gnome.Sdk',
+        'default-branch': branch,
+        'command': 'cockpit-client',
+        'rename-icon': 'cockpit-client',
+        'finish-args': [
+            '--talk-name=org.freedesktop.Flatpak',
+            '--socket=wayland',
+            '--socket=fallback-x11',
+            '--device=dri',
+            '--share=ipc'
+        ],
+        'modules': [
+            {
+                'name': 'cockpit-client',
+                'buildsystem': 'autotools',
+                'config-opts': [
+                    '--enable-cockpit-client',
+                    '--enable-pybridge',
+                    '--disable-polkit',
+                    '--disable-ssh',
+                    '--disable-pcp',
+                    '--with-systemdunitdir=/invalid',
+                    'CPPFLAGS=-Itools/mock-build-env',
+                    '--with-admin-group=root',
+                    '--disable-doc'
+                ],
+                'make-args': [
+                    'build-for-flatpak'
+                ],
+                'make-install-args': [
+                    f'DOWNSTREAM_RELEASES_XML={RELEASES_XML}'
+                ],
+                'install-rule': 'install-for-flatpak',
+                'sources': [
+                    source_info,
+                    {
+                        'type': 'file',
+                        'path': RELEASES_XML,
+                    }
+                ]
+            },
+            *extra_modules
+        ]
+    }
+
+
+def get_packages(origin: Optional[str]) -> List[Any]:
+    # 1. If --packages is explicitly given, always use it
+    # 2. Otherwise, try to use the already-existing file
+    # 3. ... but if it doesn't exist, act like 'downstream' (effectively the default)
+    # 4. In any case, write the local file if it changed (or is new)
+
+    try:
+        with open(PACKAGES_JSON, 'r') as file:
+            local_packages = json.load(file)
+    except FileNotFoundError:
+        local_packages = None
+
+    if origin == 'none':
+        packages = []
+
+    elif origin == 'upstream':
+        packages = [module_for_upstream(repo) for repo in UPSTREAM_REPOS]
+
+    elif origin == 'downstream' or local_packages is None:
+        packages = fetch_json(DOWNSTREAM_PACKAGES_URL)
+
+    else:
+        packages = local_packages
+
+    # Update local file (only if it) changed
+    if packages != local_packages:
+        write_json(PACKAGES_JSON, packages)
+
+    return packages
+
+
+def main() -> None:
+    parser = argparse.ArgumentParser()
+    parser.add_argument('--packages', choices=('none', 'upstream', 'downstream'),
+                        help="Get extra packages list from upstream/downstream")
+    parser.add_argument('--sha256', help="sha256sum of the source release")
+    parser.add_argument('location', help="Location of upstream source release (default: build a new one)", nargs='?')
```

Comment:
Shouldn't this become a proper option as well now?

Reply:
I could see doing something like `--url` and `--file` or so... but the current approach also seems fine.

The thing that was odd with the old syntax (sha256 as a second optional positional argument) is that python's `argparse`'s `--help` output was incorrect.  That's why I turned that one into a flag.  If anything, I'd put it back to how it was before, with both of them as positionals.